_posts/2019-09-29-code-quality-ruby-gems.md

```diff
@@ -0,0 +1,157 @@
+
+All of them use:
+
+- [Flog](https://github.com/seattlerb/flog)
+- [Churn](https://github.com/danmayer/churn)
+
+## RubyCritic
+
+[RubyCritic](https://github.com/whitesmith/rubycritic) is a gem that uses
+static analysis gems such as [Reek](https://github.com/troessner/reek),
+[Flay](https://github.com/seattlerb/flay),
+and [Flog](https://github.com/seattlerb/flog) to provide a quality report of
+your Ruby code.
+
+One of the best things about this tool is that it provides a quick overview of
+the project you're analyzing:
+
+<img src="/blog/assets/images/quality/ruby-critic-overview.png" alt="RubyCritic Overview for alphagov/e-petitions">
+
+With this you can get a quick glimpse of the complexity and churn in all files.
+
+In the next section you can find all the files that have an "F" grade:
+
+<img src="/blog/assets/images/quality/ruby-critic-f-files.png" alt="RubyCritic: Files that need attention">
+
+With this you can pick what files you want to refactor next. :)
+
+In the last section you can find all the code smells in the project:
+
+<img src="/blog/assets/images/quality/ruby-critic-smells.png" alt="RubyCritic: Smelly files">
+
+This section could use some improvement. It seems to sort code smells alphabetically.
+You might want to use that section if you prefer to focus on one smell at a time.
+
+You can play around with a sample report over here:
+[https://fastruby.github.io/quality/#ruby-critic](https://fastruby.github.io/quality/#ruby-critic)
+
+## MetricFu
+
+Just like RubyCritic, [MetricFu](https://github.com/metricfu/metric_fu) uses
+other Ruby gems to generate a list of reports for you:
+
+<img src="/blog/assets/images/quality/metric-fu-reports.png" alt="MetricFu: All the reports">
+
+Unlike RubyCritic, MetricFu does not provide a quick overview of the application's
+codebase. You need to drill down the reports list to investigate each aspect of
+the quality report.
+
+For instance, if you want to find what files have been updated the most, you will
+need to review the _Churn_ report:
+
+<img src="/blog/assets/images/quality/metric-fu-churn.png" alt="MetricFu: Churn report">
+
+Indeed: Files that change **a lot** in your codebase may be a bad sign.
+
+Just like RubyCritic, MetricFu generates a report using Reek:
+
+<img src="/blog/assets/images/quality/metric-fu-reek.png" alt="MetricFu: Reek report">
+
+With this report you can get a quick glimpse about the most common code smells
+in your project.
+
+If you check out the Flog section, you will find the methods that are hardest to
+test, the ones that are most complex:
+
+<img src="/blog/assets/images/quality/metric-fu-flog.png" alt="MetricFu: Flog report">
+
+MetricFu is definitely more ambitious than Attractor and RubyCritic, but it hasn't
+been actively maintained for years.
+
+You can play around with a sample report over here:
+[https://fastruby.github.io/quality/#metric-fu](https://fastruby.github.io/quality/#metric-fu)
+
+## Attractor
```

Comment:
I would add a link to the title or somewhere in the copy for Attractor, there is no link to it

Reply:
@ercohen14 done 👍